shared/libraries/config_protocol/include/config_protocol/config_protocol_client.h

```diff
@@ -325,13 +325,25 @@ void ConfigProtocolClient<TRootDeviceImpl>::reconnect()
     protocolHandshake(clientComm->getProtocolVersion());
     enumerateTypes();
 
-    const StringPtr serializedDevice = clientComm->requestSerializedRootDevice();
+    if (restoreClientConfigOnReconnect)
+    {
+        const auto serializer = JsonSerializer();
+        rootDevice.asPtr<IUpdatable>().serializeForUpdate(serializer);
+        StringPtr serializedClientRootDevice = serializer.getOutput();
+
+        const auto deserializer = JsonDeserializer();
+        deserializer.update(rootDevice.asPtr<IUpdatable>(), serializedClientRootDevice);
```

Comment:
This will automatically send the configuration to the server, correct?

Reply:
Yes, and will also receive the feedback as an `UpdateEnd` core event, to ensure the configuration match between the server and client, if the server rejects any changes.

Reply:
Ok